marklogic-langchain4j/src/main/java/com/marklogic/langchain4j/classifier/TextClassifier.java

```diff
@@ -29,8 +37,12 @@ public TextClassifier(String host, String https, String port, String endpoint, S
         Token token;
         try {
             token = tokenFetcher.getAccessToken();
+            DocumentBuilderFactory factory = DocumentBuilderFactory.newInstance();
+            builder = factory.newDocumentBuilder();
         } catch (CloudException e) {
```

Comment:
Couple things here - a DocumentBuilder isn't thread-safe. Within the context of our connector, where a partition will have a single instance of TextClassifier, that shouldn't be a problem. But it may be a problem once we shift to a UDF. And it would definitely be a problem if we move all of this logic into a separate non-Spark-related module. So just instantiate the DBF instead. 

And related to that - `DOMHelper` already has some support in it for the annoying checked exceptions thrown by the JAXP library. Use that and rename `getDocumentBuilder` to `newDocumentBuilder` and use that instead, as it already has error handling built in.

Reply:
I took a shot. I'm not sure I have what you're asking for.

Reply:
I actually couldn't figure out what you wanted with DOMHelper, so I'm just building the DBF at instantiation and creating a new Builder during processing.